thirdeye/thirdeye-pinot/src/test/java/org/apache/pinot/thirdeye/datasource/resources/TestAnomalyFlattenResource.java

```diff
@@ -63,53 +66,87 @@ void afterClass() {
+    }
+    List<Map<String, Object>> actualResults = AnomalyFlattenResource.reformatDataFrameAndAnomalies(metricDataFrame,
+        anomalies, Arrays.asList("what", "where", "when", "how"));
     List<Map<String, Object>> expectedResults = Arrays.asList(
         new HashMap<String, Object>() {
           {
-            put(ANOMALY_ID, 2L);
             put("what", "a");
             put("where", "b");
             put("when", "c");
             put("how", "d");
-            put("metric", 0d);
-            put(ANOMALY_COMMENT, "");
+            put("metric", 0.1d);
+            put(ANOMALY_COMMENT, new HashMap<Long, String>(){{
+              put(1l, "#1 is Not Reviewed Yet");
+            }});
           }
         },
         new HashMap<String, Object>() {
           {
-            put(ANOMALY_ID, 3L);
             put("what", "e");
             put("where", "f");
             put("when", "g");
             put("how", "h");
-            put("metric", 0d);
-            put(ANOMALY_COMMENT, "");
+            put("metric", 0.2d);
+            put(ANOMALY_COMMENT, new HashMap<Long, String>(){{
+              put(2l, "#2 is Not Reviewed Yet");
+            }});
           }
         }
     );
     Assert.assertEquals(actualResults, expectedResults);
   }
 
   @Test
-  public void testFlatAnomalyResult() {
+  public void testExtractComments() {
     List<MergedAnomalyResultDTO> anomalies = TestMergedAnomalyResultManager.mockAnomalies(detectionConfigId);
-    MergedAnomalyResultDTO anomaly = anomalies.get(0);
-    anomaly.setId(1L);
-    Map<String, Object> actualMap = AnomalyFlattenResource.flatAnomalyResult(anomaly, null);
-    Map<String, Object> expectMap = new HashMap<String, Object>() {
+    for (int i = 0; i < anomalies.size(); i++) {
+      anomalies.get(i).setId((long) i + 1);
+    }
+    Map<DimensionMap, Map<Long, String>> actualMap = AnomalyFlattenResource.extractComments(anomalies);
+    Map<DimensionMap, Map<Long, String>> expectMap = new HashMap<DimensionMap, Map<Long, String>>() {
       {
-        put(ANOMALY_ID, anomaly.getId());
-        put("what", "a");
-        put("where", "b");
-        put("when", "c");
-        put("how", "d");
-        put("metric", 0d);
-        put(ANOMALY_COMMENT, "");
+        DimensionMap dm = new DimensionMap();
+        dm.put("what", "a");
+        dm.put("where", "b");
+        dm.put("when", "c");
+        dm.put("how", "d");
+        put(dm, new HashMap<Long, String>(){
+          {
+            put(1l, "#1 is Not Reviewed Yet");
+          }});
+
+        dm = new DimensionMap();
+        dm.put("what", "e");
+        dm.put("where", "f");
+        dm.put("when", "g");
+        dm.put("how", "h");
+        put(dm, new HashMap<Long, String>(){
+          {
+            put(2l, "#2 is Not Reviewed Yet");
+          }});
       }
     };
     Assert.assertEquals(actualMap, expectMap);
   }
+
+  private DataFrame mockDataFrame() {
+    StringSeries what = StringSeries.buildFrom("a", "e");
```

Comment:
remove these Series defined here if not used.

Reply:
Done. Thanks